dependencies.yaml

```diff
@@ -96,8 +96,8 @@ dependencies:
       - output_types: [conda, requirements]
         packages:
           - cupy >=12.0.0
-          - numba >=0.49.0
-          - numpy >=1.19,<1.24
+          - numba >=0.57.0
+          - numpy >=1.19
```

Comment:
I think we're pinning `numpy >=1.21` in most other RAPIDS packages.

Reply:
Since this was already setting a lower bound of NumPy 1.19, went ahead and left it. No strong feelings though. Will defer to maintainers here

Reply:
I'm happy to leave it as-is, too -- just wanted to give some context for why that version is relevant.

Reply:
Yeah Numba 0.57.0 is needed for CUDA 12 support

Ah ok. That's a good point. Let's bump NumPy to 1.21 then

Reply:
```suggestion
          - numpy >=1.21
```